### ISSUE TITLE
Sumneko warning

### ISSUE THREAD

**Issue by rgnkn** [monaqa/dial.nvim#46]
With the folke's new `neodev.nvim` I receive several warnings with my configuration. As I assume that the warnings are not "serious" and that everything runs just fine, I just silenced them like this.

```
{
    "monaqa/dial.nvim",
    event = "BufRead",
    config = function()
        local augend = require("dial.augend")
        require("dial.config").augends:register_group(
            {
                default = {
                    augend.integer.alias.decimal,
                    augend.integer.alias.hex,
                    augend.integer.alias.binary,
                    ---@diagnostic disable-next-line
                    augend.date.alias["%Y/%m/%d"],
                    ---@diagnostic disable-next-line
                    augend.date.alias["%d.%m.%y"],
                    augend.semver.alias.semver
                }
            }
        )
    end
}
```

Without the "silencing" I receive the following warning on each of the two lines:

```
Cannot assign `AugendDate` to `Augend`.- `AugendDate` cannot match `Augend`- Type `AugendDate` cannot match `Augend` Lua Diagnostics. (assign-type-mismatch) [390, 21]
```

It might be a good idea if you could check this "warning" if it has any bad consequences.

Feel free to close this issue, if you think that this isn't important.

**Comment by yhu266** [comment]
This is not a significant issue. `AugendConstant` is an implementation of `Augend`, and the later is just a placeholder name, not an actual class.

https://github.com/monaqa/dial.nvim/blob/f6d274bd8e32be65b99f1052a8b5353ef5e702d7/lua/dial/augend/constant.lua#L6-L9

https://github.com/monaqa/dial.nvim/blob/fa74b74732520e224ed05cb8bc938806258011ce/lua/dial/types.lua#L15

**Comment by rgnkn** [comment]
Yep, as mentioned I already assumed that there is no big problem with this and therefore tried to phrase this issue as passively as possible.

Anyway, I prefer disabling the warnings on the spot(s) instead of silencing them altogether.
